pydantic/validators.py

```diff
@@ -409,9 +420,14 @@ def check(self, config: Type['BaseConfig']) -> bool:
 ]
 
 
-def find_validators(type_: AnyType, config: Type['BaseConfig']) -> Generator[AnyCallable, None, None]:
+def find_validators(  # noqa: C901 (ignore complexity)
+    type_: AnyType, config: Type['BaseConfig']
+) -> Generator[AnyCallable, None, None]:
     if type_ is Any:
         return
+    if getattr(type_, '__origin__', None) is Literal:
```

Comment:
Can we combine this check with `is_callable_type` which is called later and also does `getattr(type_, '__origin__', None)`.

Reply:
also makes sense to move this check down to after `if type_ is Pattern` since it'll rarely be true.

Reply:
we probably want to reorganise this function completely but that's another PR.

Reply:
I attempted to address this; not sure if it's as "combined" as you'd like yet